packages/domain-picker/src/domain-picker/index.tsx

```diff
@@ -170,29 +160,67 @@ const DomainPicker: FunctionComponent< Props > = ( {
 						</div>
 					) }
-					<div className="domain-picker__suggestion-item-group">
-						{ domainSuggestions?.map( ( suggestion, i ) => (
-							<SuggestionItem
-								key={ suggestion.domain_name }
-								suggestion={ suggestion }
-								railcarId={ baseRailcarId ? `${ baseRailcarId }${ i }` : undefined }
-								isRecommended={ i === 0 }
-								onRender={ () =>
-									handleItemRender( suggestion, `${ baseRailcarId }${ i }`, i, i === 0 )
-								}
-								onSelect={ onDomainSelect }
-							/>
-						) ) ?? times( quantity, ( i ) => <SuggestionItemPlaceholder key={ i } /> ) }
-						{ ! isExpanded &&
-							allDomainSuggestions?.length &&
-							allDomainSuggestions?.length > quantity && (
-								<div className="domain-picker__show-more">
-									<Button onClick={ () => setIsExpanded( true ) }>
-										{ __( 'View more results' ) }
-									</Button>
+					{ domainSuggestions ? (
+						<div className="domain-picker__suggestion-sections">
+							<div className="domain-picker__suggestion-item-group">
+								<p className="domain-picker__suggestion-group-label">{ __( 'Sub-domain' ) }</p>
+								{ domainSuggestions[ 0 ] && (
+									<SuggestionItem
+										key={ domainSuggestions[ 0 ].domain_name }
+										suggestion={ domainSuggestions[ 0 ] }
+										railcarId={ baseRailcarId ? `${ baseRailcarId }1` : undefined }
+										onRender={ () =>
+											handleItemRender( domainSuggestions[ 0 ], `${ baseRailcarId }1`, 0, false )
+										}
+										onSelect={ onDomainSelect }
+									/>
+								) }
+							</div>
+
+							{ domainSuggestions.length > 1 && (
+								<div className="domain-picker__suggestion-item-group">
+									<p className="domain-picker__suggestion-group-label">
+										{ __( 'Custom domains' ) }
+									</p>
+									{ domainSuggestions.slice( 1 ).map( ( suggestion, i ) => (
+										<SuggestionItem
+											key={ suggestion.domain_name }
+											suggestion={ suggestion }
+											railcarId={ baseRailcarId ? `${ baseRailcarId }${ i }` : undefined }
+											isRecommended={ i === 0 }
+											onRender={ () =>
+												handleItemRender( suggestion, `${ baseRailcarId }${ i }`, i, i === 0 )
+											}
+											onSelect={ onDomainSelect }
+										/>
+									) ) }
 								</div>
 							) }
-					</div>
+
+							{ ! isExpanded &&
+								allDomainSuggestions?.length &&
+								allDomainSuggestions?.length > quantity && (
+									<div className="domain-picker__show-more">
+										<Button onClick={ () => setIsExpanded( true ) }>
+											{ __( 'View more results' ) }
+										</Button>
+									</div>
+								) }
+						</div>
+					) : (
+						<div className="domain-picker__suggestion-sections">
+							<div className="domain-picker__suggestion-item-group">
+								<p className="domain-picker__suggestion-group-label">{ __( 'Sub-domain' ) }</p>
```

Comment:
This duplication of markup could be avoided if we'd replace the `&&` with a ternary at https://github.com/Automattic/wp-calypso/pull/43830/files#diff-f948bcd10a91168abd35256bd0ae22aaR167 and render the placeholder there.